app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,109 @@
+class @ProjectFindFile
+  constructor: (@options)->
+    @filePaths = {}
+    @element = ".file-finder-input"
+    @row = "<tr class='tree-item'><td class='tree-item-file-name'><i class='fa fa-file-text-o fa-fw'></i><span class='str-truncated'><a href=%blobItemUrl>%filePath</a></span></td></tr>"
+
+    # init key event(shortcut, find file)
+    @initKeyEvent()
+
+    # focus text input box
+    $(@element).focus()
+
+    # load file list
+    @load(@options.url)
+
+  # init key event
+  initKeyEvent: ->
+    # unbind and bind keyup/keydown event at text input box
+    $(document).off "keyup"
+    $(document).off "keyup", @element
+    $(document).off "keydown", "body"
+
+    # $(document).on "keyup", @goBack
+    $(document).on "keyup", (event) =>
+      if event.keyCode == 27
+        event.preventdefault
+        location.href = @options.treeUrl
+      return false
+
+    $(document).on "keyup", @element, (event) =>
+      keyCode = event.keyCode if event
+      # up down right left arrow(37~40) esc(27) caps(20) ctrl(17) alt(18) shift(16) enter(13) cmd(91)
+      if (keyCode < 37 || keyCode > 40) && keyCode != 20 && keyCode != 17 && keyCode != 18 && keyCode != 16 && keyCode != 13 && keyCode != 91

```

Comment:
We already use moustrap (https://github.com/ccampbell/mousetrap), is this keycode checking necessary with that?